src/main/java/ca/uhn/fhir/jpa/starter/service/NotificationService.java

```diff
@@ -98,6 +101,43 @@ public void deletePreviousRecords() {
 		datasource.deleteRecordsByTimePeriod(previousDate);
 	}
 
+	private void updateAppointmentSMSInfoRecord(ComGenerator comGenerator){
+		IGenericClient fhirClient = FhirClientAuthenticatorService.getFhirClient();
+		String resourceId = comGenerator.getResourceId();
+		String patientId = comGenerator.getPatientId();
+		String patientCardNumber = FhirUtils.getPatientCardNumberByPatientId(patientId, fhirClient);
+		NotificationDataSource dataSource = NotificationDataSource.getInstance();
+		List<SMSInfo> smsInfo = dataSource.fetchSMSRecordsByResourceId(resourceId);
+		Pair<String, String> encIdAndOrgIdPair = FhirUtils.getEncounterIdAndOrganizationIdForAppointment(resourceId, fhirClient);
+		if (!smsInfo.isEmpty() && encIdAndOrgIdPair != null){
+			SMSInfo smsInfoRecord = smsInfo.get(0);
+			smsInfoRecord.setStatus(comGenerator.getCommunicationStatus());
+			smsInfoRecord.setPatientCardNumber(patientCardNumber);
+			smsInfoRecord.setEncounterId(encIdAndOrgIdPair.getFirst());
+			smsInfoRecord.setOrganizationId(encIdAndOrgIdPair.getSecond());
+			if (Objects.equals(comGenerator.getCommunicationStatus(), MessageStatus.SENT.name()))
+				smsInfoRecord.setSentAt(comGenerator.getUpdatedAt());
+			dataSource.update(smsInfoRecord);
+		}
+	}
+
+	private void updateEncounterSMSInfoRecord(ComGenerator comGenerator){
+		IGenericClient fhirClient = FhirClientAuthenticatorService.getFhirClient();
+		String resourceId = comGenerator.getResourceId();
+		String patientId = comGenerator.getPatientId();
+		String patientCardNumber = FhirUtils.getPatientCardNumberByPatientId(patientId, fhirClient);
+		NotificationDataSource dataSource = NotificationDataSource.getInstance();
+		List<SMSInfo> smsInfo = dataSource.fetchSMSRecordsByResourceId(resourceId);
+		if (!smsInfo.isEmpty()){
+			SMSInfo smsInfoRecord = smsInfo.get(0);
+			smsInfoRecord.setStatus(comGenerator.getCommunicationStatus());
+			smsInfoRecord.setPatientCardNumber(patientCardNumber);
+			if (Objects.equals(comGenerator.getCommunicationStatus(), MessageStatus.SENT.name()))
+				smsInfoRecord.setSentAt(comGenerator.getUpdatedAt());
+			dataSource.update(smsInfoRecord);
+		}
+	}
```

Comment:
Expect for adding organization id, both of these function doing the same task. Can you make it a single function?

```
if (comGenerator.getResourceType == "Appointment") {
    Pair<String, String> encIdAndOrgIdPair = FhirUtils.getEncounterIdAndOrganizationIdForAppointment(resourceId, fhirClient);
    smsInfoRecord.setEncounterId(encIdAndOrgIdPair.getFirst());
			smsInfoRecord.setOrganizationId(encIdAndOrgIdPair.getSecond());
}
```